src/integration_tests/integration_utils.py

```diff
@@ -26,20 +26,60 @@ def check_function(
     if not kwargs:
         latexified = frontend.function(fn)
         assert str(latexified) == latex
-        assert latexified._repr_latex_() == rf"$$ \displaystyle {latex} $$"
+        assert latexified._repr_latex_() == r"$$ \displaystyle " + latex + " $$"
```

Comment:
Restore the original expression, and don't use "+" concatenation as well as possible. f-strings or `join()` are basically the most fastest option to concatenate strings and "+" is the worst choice.

Ditto for all other places.

```python
def f():
  for i in range(1_000_000):
    x = f"The answer is {i}."
  return x

def g():
  for i in range(1_000_000):
    x = "The answer is " + str(i) + "."
  return x

%time f()
%time g()
```

```
CPU times: user 191 ms, sys: 0 ns, total: 191 ms
Wall time: 192 ms
CPU times: user 327 ms, sys: 0 ns, total: 327 ms
Wall time: 327 ms
```

Reply:
Good to know, I think I've converted it all back. When codegen though lots of places can't use `f` strings because of `\` not allowed :(